be/src/olap/reader.cpp

```diff
@@ -613,15 +613,14 @@ Status TabletReader::_init_delete_condition(const ReaderParams& read_params) {
     }
 
     auto delete_init = [&]() -> Status {
-        return _delete_handler.init(_tablet_schema, _tablet->delete_predicates(),
+        return _delete_handler.init(_tablet_schema, read_params.delete_predicates,
                                     read_params.version.second, this);
     };
 
     if (read_params.reader_type == READER_ALTER_TABLE) {
         return delete_init();
     }
 
-    std::shared_lock rdlock(_tablet->get_header_lock());
```

Comment:
Did this lock have any impact before?

Reply:
`delete_init()` access `tablet->delete_predicates()` in previous code, the access should be protected by a read lock, and now we don't need to access `tablet->delete_predicates()` here any more, so the lock should be removed.